python-main.js

```diff
@@ -18,12 +18,35 @@ function pythonEditor(id) {
     editor.fontSizeStep = 4;
 
     // Represents the ACE based editor.
+    var langTools = ace.require("ace/ext/language_tools");
     var ACE = ace.edit(id);  // The editor is in the tag with the referenced id.
     ACE.setOptions({
-        enableSnippets: true  // Enable code snippets.
+        enableSnippets: true,  // Enable code snippets.
+        enableBasicAutocompletion: true, // Enable (automatic) autocompletion
+        enableLiveAutocompletion: true
     });
-    ACE.setTheme("ace/theme/kr_theme");  // Make it look nice.
-    ACE.getSession().setMode("ace/mode/python");  // We're editing Python.
+    ACE.$blockScrolling = Infinity; // Silences the 'blockScrolling' warning
+
+    var horizontalWordList = populateWordList();
+
+    var staticWordCompleter = {
+        identifierRegexps: [/[a-zA-Z_0-9\.\-\u00A2-\uFFFF]/],
+        getCompletions: function(editor, session, pos, prefix, callback) {
+            var wordList = horizontalWordList;
+            
+            callback(null, wordList.map(function(word) {
+                return {
+                    caption: word,
+                    value: word,
+                    meta: "static"
+                };
+            }));
+        }
+    }
```

Comment:
There is a missing semicolon, but there is no need to add a new commit just for this, if this branch is updated in the future just remember to add this one as well.